src/librustc_typeck/check/expr.rs

```diff
@@ -1440,6 +1440,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         display
     }
 
+    fn complain_about_invalid_type_params(&self, args: Option<Span>) {
```

Comment:
Please add tests for each situation in which this method is called so the method is exercised.